app/PaymentDrivers/Mollie/IDEAL.php

```diff
@@ -91,48 +80,70 @@ public function paymentView(array $data)
                     'gateway_type_id' => GatewayType::IDEAL,
                     'payment_type_id' => PaymentType::IDEAL,
                 ],
-            ]);
+            ];
+
+            if ($this->mollie->company_gateway->token_billing == 'always') {
+                // Check if a mollie CustomerId already exists for this client, if so, use that
+                $gateway_customer_reference = null;
+                if ($this->mollie->client->gateway_tokens->count() > 0) {
+                    $gateway_customer_reference = $this->mollie->client->gateway_tokens->first()->gateway_customer_reference;
+                }
+                if (!$gateway_customer_reference) {
+                    $customer = $this->mollie->gateway->customers->create([
+                        'name' => $this->mollie->client->name,
+                        'email' => $this->mollie->client->present()->email(),
+                        'metadata' => [
+                            'id' => $this->mollie->client->hashed_id,
+                        ],
+                    ]);
+                    $gateway_customer_reference = $customer->id;
+                }
+
+                $data['customerId'] = $gateway_customer_reference;
+                $data['sequenceType'] = 'first';
+
+                $this->mollie->payment_hash
+                    ->withData('mollieCustomerId', $gateway_customer_reference)
+                    ->withData('shouldStoreToken', true);
+            }
+
+            try {
+                $payment = $this->mollie->gateway->payments->create($data);
+            } catch (ApiException $e) {
+                if (preg_match('/method selected.*not accept recurring payments/', $e->getMessage())){
+                    // Can't create mandate, log
+                    SystemLogger::dispatch(
+                        ['error' => 'Could not create mandate upon iDeal payment. Check if all required payment methods are enabled in Mollie (e.g. SEPA Direct Debit).', 'exception' => $e],
```

Comment:
Instead of logging the entire exception, just $e->getMessage() will suffice here.